internal/stm32CubeMX/mxDevice.go

```diff
@@ -487,6 +490,35 @@ func getMCIMode(fMain *os.File, peripheral string) (string, error) {
 	return "", nil
 }
 
+// Get MMC Freq
+func getMMCFreq(contextMap map[string]map[string]string, peripheral string) string {
+	var freq string
+
+	if !strings.HasPrefix(peripheral, "SDMMC") && !strings.HasPrefix(peripheral, "SDIO") {
+		return ""
+	}
+
+	freq = getUserConstant(contextMap, peripheral+"_PERIPH_CLOCK_FREQ")
+	if freq != "" {
+		// Frequency defined by user in CubeMX
+		return freq
+	}
+
+	peri := contextMap["RCC"]
+
+	if len(peri) > 0 {
+		periphStrings := [2]string{peripheral, strings.TrimRight(peripheral, getDigitAtEnd(peripheral))}
+		for _, ps := range periphStrings {
+			for p, freq := range peri {
+				if strings.HasPrefix(p, ps) && strings.Contains(p, "Freq_Value") {
```

Comment:
This looks to me very similar to I2C and SPI, but the part with "Freq_Value" and getDigitAtEnd is missing here. Is certainly correct, just guessing...

Reply:
Yes. Unlike I2C and SPI, SDIO doesn't have a peripheral index (like SDIO1) when there's only one SDIO/SDMMC peripheral in the STM32 device. This slight difference accounts for the variation in parsing.